File: src/main/java/arcana/common/research/sections/TextSection.java
package arcana.common.research.sections;

import arcana.common.research.BookSection;
import com.google.gson.JsonObject;
import net.minecraft.util.text.TranslationTextComponent;

public class TextSection extends BookSection {
    TranslationTextComponent text;
    @Override
    public void deserialize(JsonObject json) {
        text = new TranslationTextComponent(json.get("content").getAsString());
    }
}

File: src/main/kotlin/arcana/common/research/sections/TextSection.kt
package arcana.common.research.sections

import arcana.common.research.BookSection
import com.google.gson.JsonObject
import net.minecraft.util.text.TranslationTextComponent

class TextSection : BookSection() {
    lateinit var text: TranslationTextComponent
    override fun deserialize(json: JsonObject) {
        text = TranslationTextComponent(json["content"].asString)
    }
}